tensorflow/python/keras/layers/local_test.py

```diff
@@ -20,216 +20,226 @@
 
 import numpy as np
 
+from absl.testing import parameterized
+
 from tensorflow.python import keras
 from tensorflow.python.framework import test_util as tf_test_util
 from tensorflow.python.keras import testing_utils
 from tensorflow.python.platform import test
 from tensorflow.python.training.rmsprop import RMSPropOptimizer
 
 
+_DATA_FORMAT_PADDING_IMPLEMENTATION = [
+    {"data_format": 'channels_first', "padding": 'valid', "implementation": 1},
+    {"data_format": 'channels_first', "padding": 'same', "implementation": 1},
+    {"data_format": 'channels_last', "padding": 'valid', "implementation": 1},
+    {"data_format": 'channels_last', "padding": 'same', "implementation": 1},
+    {"data_format": 'channels_first', "padding": 'valid', "implementation": 2},
+    {"data_format": 'channels_first', "padding": 'same', "implementation": 2},
+    {"data_format": 'channels_last', "padding": 'valid', "implementation": 2},
+    {"data_format": 'channels_last', "padding": 'same', "implementation": 2}]
+
+
 @tf_test_util.run_all_in_graph_and_eager_modes
-class LocallyConnected1DLayersTest(test.TestCase):
+class LocallyConnected1DLayersTest(test.TestCase, parameterized.TestCase):
 
-  def test_locallyconnected_1d(self):
+  @parameterized.parameters(_DATA_FORMAT_PADDING_IMPLEMENTATION)
+  def test_locallyconnected_1d(self, data_format, padding, implementation):
     with self.cached_session():
       num_samples = 2
       num_steps = 8
       input_dim = 5
       filter_length = 3
       filters = 4
 
-      for padding in ['valid', 'same']:
-        for strides in [1]:
-          if padding == 'same' and strides != 1:
-            continue
-          for data_format in ['channels_first', 'channels_last']:
-            for implementation in [1, 2]:
-              kwargs = {
-                  'filters': filters,
-                  'kernel_size': filter_length,
-                  'padding': padding,
-                  'strides': strides,
-                  'data_format': data_format,
-                  'implementation': implementation
-              }
-
-              if padding == 'same' and implementation == 1:
-                self.assertRaises(ValueError,
-                                  keras.layers.LocallyConnected1D,
-                                  **kwargs)
-              else:
-                testing_utils.layer_test(
-                    keras.layers.LocallyConnected1D,
-                    kwargs=kwargs,
-                    input_shape=(num_samples, num_steps, input_dim))
-
-  def test_locallyconnected_1d_regularization(self):
+      for strides in [1]:
```

Comment:
Is there a point to leaving this for loop & the continue if we're cleaning up this file anyway?

Reply:
@tomerk 

No, just keep the existing logic without much changes in parameterized parameter.